### ISSUE TITLE
Support OS X

### ISSUE THREAD

**Issue by p-e-w** [p-e-w/maybe#3]

**Comment by codeskyblue** [comment]
+1

**Comment by viorelsfetea** [comment]
Hi Philipp,

This is something that I would love to get my hands dirty with. Could you share your findings so far, please?

Thank you,
Viorel

**Comment by GabLeRoux** [comment]
Yeah would be nice
```
maybe rm -r app
```
currently explodes with
```
Warning: Running without traceFork support. Syscalls from subprocesses can not be intercepted.
Traceback (most recent call last):
  File "/usr/local/bin/maybe", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/maybe/maybe.py", line 154, in main
    prepareProcess(process)
  File "/usr/local/lib/python2.7/site-packages/maybe/maybe.py", line 53, in prepareProcess
    process.syscall()
  File "/usr/local/lib/python2.7/site-packages/ptrace/debugger/process.py", line 469, in syscall
    ptrace_syscall(self.pid, signum)
  File "/usr/local/lib/python2.7/site-packages/ptrace/binding/func.py", line 254, in ptrace_syscall
    ptrace(PTRACE_SYSCALL, pid, 1, signum)
  File "/usr/local/lib/python2.7/site-packages/ptrace/binding/func.py", line 148, in ptrace
    raise PtraceError(message, errno=errno, pid=pid)
ptrace.error.PtraceError: ptrace(cmd=22, pid=19257, 1, 0) error #22: Invalid argument
```

**Comment by OJFord** [comment]
@viorelsfetea The work to be done is on `python-ptrace`, [which needs to support the ktrace back-end](http://python-ptrace.readthedocs.org/en/latest/todo.html). It's beyond me to be of much use though unfortunately.

I don't know how much of a priority that is either - it seems to have been on the todo list for at least a couple of years; the last commit was in November. Once done though, I suppose OS X support should come to `maybe` more or less for free.

**Comment by iBobik** [comment]
Then it will be nice to maintain formula in http://brew.sh package manager. :-)

**Comment by ylluminarious** [comment]
+1

**Comment by OJFord** [comment]
Can supporters please 'subscribe' to updates with the button on the right rather than commenting; filling our inboxes with '+1's?

With a repo as popular as this, the popularity of OS X, and the fact that the issue was actually started by the repo owner, I think it's unlikely that statements of support is what's required to get this done any faster.

**Comment by ylluminarious** [comment]
@OJFord Sorry, didn't mean to cause any frustration.

**Comment by strages** [comment]
Some better explanation of why it doesn't currently work on osx would be appreciated!

**Comment by ChrisJohnRyan** [comment]
@strages 
They commented on it here 
"The work to be done is on python-ptrace, which needs to support the ktrace back-end. It's beyond me to be of much use though unfortunately.

I don't know how much of a priority that is either - it seems to have been on the todo list for at least a couple of years; the last commit was in November. Once done though, I suppose OS X support should come to maybe more or less for free." @OJFord

**Comment by strages** [comment]
Alright^^

On Thu, Feb 18, 2016, 5:53 PM Christopher John Ryan <
notifications@github.com> wrote:

> @strages <https://github.com/strages>
> They commented on it here
> "The work to be done is on python-ptrace, which needs to support the
> ktrace back-end. It's beyond me to be of much use though unfortunately.
>
> I don't know how much of a priority that is either - it seems to have been
> on the todo list for at least a couple of years; the last commit was in
> November. Once done though, I suppose OS X support should come to maybe
> more or less for free." @OJFord <https://github.com/OJFord>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/p-e-w/maybe/issues/3#issuecomment-185813060>.
>
-- 
--

*Dimitrie Hoekstra, UI & Interaction Designer*
dhesign.com  | dimitriehoekstra@gmail.com | LinkedIn
<https://www.linkedin.com/in/dimitrieh> | Twitter
<https://twitter.com/strages33/>

**Comment by rbrich** [comment]
I don't think ktrace works on OSX anymore. It was replaced by dtrace. And dtrace was rendered unusable in OSX 10.11.

You can try to disable the System Integrity Protection (which blocks dtrace), but even then dtrace is not allowed to probe binaries installed in system directories. That means no dtrace probes on /bin/rm etc. You can still watch direct system calls made by a binary in your home dir, but when it calls something like /bin/rm, then you are out of luck. And, to make even this work, you must first reboot to Recovery mode and disable SIP through the console. Which really isn't user friendly at all.

In short, there is sadly no feasible way to make this utility (or python-ptrace) work on recent OS X.

Some details here: http://apple.stackexchange.com/questions/193368/what-is-the-rootless-feature-in-el-capitan-really

**Comment by OJFord** [comment]
@rbich Damn. I'd forgotten about that. Ironically, to quote myself from that very page:

 > So, probably the answer is "most of the time you won't notice; when you do, you'll notice hard."

**Comment by strages** [comment]
same problem finder replacements such as totalfinder and xtrafinder :P
sooooo probability of this ever getting to osx?

On Mon, Feb 22, 2016 at 4:35 PM Ollie Ford <notifications@github.com> wrote:

> @rbich Damn. I'd forgotten about that. Ironically, to quote myself from
> that very page:
>
> So, probably the answer is "most of the time you won't notice; when you
> do, you'll notice hard."
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/p-e-w/maybe/issues/3#issuecomment-187232812>.
>
-- 
--

*Dimitrie Hoekstra, UI & Interaction Designer*
dhesign.com  | dimitriehoekstra@gmail.com | LinkedIn
<https://www.linkedin.com/in/dimitrieh> | Twitter
<https://twitter.com/strages33/>

**Comment by Ruben62** [comment]
Sorry guys don't know what the hell I was thinking? I'm not from the geek
squad, I was introduced to different gadgets and private browsers, tools
and addons, for my phone and PC, looks like I'm going straight, should of
asked for online help awhile back, maybe I wouldn't have wasted my money
and people like yourselves time? Sorry for the inconveniences, Freedom of
the Press, or however that goes? Could of been a star?

**Comment by p-e-w** [comment]
Looks as if OS X support isn't going to happen in the end. python-ptrace is unmaintained upstream and is therefore unlikely to support OS X, ever. :disappointed:
